### PR TITLE
add type definition file for co-util

### PR DESCRIPTION
**Pull request by codeorg**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by codeorg** [comment]
co-uitl

**Review by codeorg — COMMENTED** [review]
please add co-util to @types

File: types/co-util/tslint.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{ "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        "array-type": false,
```
Comment (gabritto) [inline types/co-util/tslint.json:3]: It doesn't look like disabling these rules is strictly necessary. Could you instead remove these rules and fix the lint errors in your declaration files? If you have a problem with how to fix a specific lint error, please let me know and I can try to help.

**Review by gabritto — CHANGES_REQUESTED** [review]

**Comment by gabritto** [comment]
On a second glance, it looks like the original library ships its own type definitions (https://unpkg.com/browse/co-util@2.0.1/lib/index.d.ts), so I believe you should be using the type definitions in the original package, instead of adding type definitions here on DT. Because of that, I'm closing this PR. But if you think there is a reason why there needs to be type definitions for this package here in DT, let me know.

**Comment by gabritto** [comment]
On a second glance, it looks like the original library ships its own type definitions (https://unpkg.com/browse/co-util@2.0.1/lib/index.d.ts), so I believe you should be using the type definitions in the original package, instead of adding type definitions here on DT. Because of that, I'm closing this PR. But if you think there is a reason why there needs to be type definitions for this package here in DT, let me know.

**Pull request closed without merging**